custom/icds_reports/reports/poshan_progress_dashboard_data.py

```diff
@@ -139,9 +149,7 @@ def prepare_quarter_dict(data, data_period, unique_id):
     return data
 
 
-def calculate_comparative_rows(data, aggregation_level, data_period, unique_id):
-    if data_period == 'quarter':
-        data = prepare_quarter_dict(data, data_period, unique_id)
+def calculate_comparative_rows(data, aggregation_level):
```

Comment:
- Instead of fetching three months in one go we are fetching month by month (added the reason in the PR description )
- data fetching is shifted to a more generic functions (fetch_month_data and fetch_quarter_data)
- Moreover, removed the aggregated query in case of the month (to make things generic as well as consistent to PPR)